pkg/http/server.go

```diff
@@ -163,6 +167,7 @@ func (s *server) StartNonBlocking() error {
 	// Start the public HTTP server
 	if s.config.PublicPort != nil {
 		publicR := s.getRouter()
+		s.useContextSetup(publicR)
 		s.useTracing(publicR)
 		s.useMetrics(publicR)
```

Comment:
This has been kept for backwards-compat, and it was checked by an E2E test, but should the public API server (used by K8s healthchecks primarily) still use metrics and tracing?

Not sure if it's needed to collect metrics on that. And it may be odd to have metrics for the "regular" HTTP server intermixed with metrics from the public HTTP server